compiler/src/iree/compiler/InputConversion/Common/ImportMLProgram.cpp

```diff
@@ -0,0 +1,118 @@
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+namespace {
+
+struct ImportMLProgramPass : public ImportMLProgramBase<ImportMLProgramPass> {
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<IREE::Util::UtilDialect>();
+  }
+  void runOnOperation() override;
+};
+
+class IREETypeConverter : public TypeConverter {
+ public:
+  IREETypeConverter();
+};
+
+// Generic 1:1 conversion pattern which effectively just renames an op.
+// It does not support regions or ops with successors.
+class OneToOneConversionPattern : public ConversionPattern {
+ public:
+  OneToOneConversionPattern(TypeConverter &converter, StringRef srcName,
+                            StringRef targetName, MLIRContext *context,
+                            PatternBenefit benefit)
+      : ConversionPattern(converter, srcName, benefit, context),
+        targetName(targetName) {}
+  LogicalResult matchAndRewrite(
+      Operation *srcOp, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    SmallVector<Type> resultTypes;
+    if (failed(typeConverter->convertTypes(srcOp->getResultTypes(),
+                                           resultTypes))) {
+      return srcOp->emitError()
+             << "could not convert result types to IREE internal types";
+    }
+
+    OperationState state(srcOp->getLoc(), targetName, operands, resultTypes,
+                         srcOp->getAttrs());
+    Operation *targetOp = rewriter.create(state);
+    rewriter.replaceOp(srcOp, targetOp->getResults());
+    return success();
+  }
+
+ private:
+  StringRef targetName;
+};
+
+class MLProgramGlobalOpPattern
+    : public OpConversionPattern<ml_program::GlobalOp> {
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult matchAndRewrite(
+      ml_program::GlobalOp srcOp, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    Type newType = typeConverter->convertType(srcOp.getType());
+    if (!newType) return failure();
+    auto globalOp = rewriter.replaceOpWithNewOp<IREE::Util::GlobalOp>(
+        srcOp, srcOp.getName(), srcOp.getIsMutable(), newType,
+        srcOp.getValue());
+    globalOp.setVisibility(srcOp.getVisibility());
+    return success();
+  }
+};
+
+}  // namespace
+
+IREETypeConverter::IREETypeConverter() {
+  addConversion([](Type t) { return t; });
+}
+
+void ImportMLProgramPass::runOnOperation() {
+  auto &context = getContext();
+  RewritePatternSet patterns(&getContext());
+  ConversionTarget target(getContext());
+  target.addLegalDialect<IREE::Util::UtilDialect>();
+  target.addIllegalDialect<ml_program::MLProgramDialect>();
+  target.markUnknownOpDynamicallyLegal([](mlir::Operation *) { return true; });
+
+  IREETypeConverter typeConverter;
+  patterns.insert<MLProgramGlobalOpPattern>(typeConverter, &getContext(), 0);
+
+  PatternBenefit specific_benefit = 100;
```

Comment:
Is this just out of caution or any reason? 1 seems to be sufficient here as there are no overlapping patterns.